com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -759,6 +760,14 @@ protected NetworkObject GetNetworkObject(ulong networkId)
         /// </summary>
         public virtual void OnDestroy()
         {
+            if (NetworkObject != null && NetworkObject.IsSpawned && IsSpawned)
```

Comment:
Do we care if `IsSpawned == true`? Isn't it only `NetworkObject.IsSpawned` that we care about since that's the object whose list we have to fix-up?